### ISSUE TITLE
Default hyperlink colour falls below min contrast ratio for AAA level for normal text (WCAG Level AAA)

### ISSUE THREAD

**Issue by jrbarnes9** [ONSdigital/design-system#2249]
`$color-ocean-blue` on white has a contrast ratio of 6.63:1 which fails WCAG Level AAA – which requires a min ratio of 7:1 for text under 18.66px

Would could asking branding to darken this colour by just 1% lightness, which would then pass this requirement.

(Page 74-75 of June 2022 DAC DS report...)

---

## Colour Contrast [Text Enhanced]
Some of the colour combinations found on the service do not meet enhanced contrasts and are likely to be difficult for people with low vision to read.

WCAG Reference:
1.4.6 Contrast (Enhanced) (Level AAA)
Understanding Contrast (Enhanced) |How to Meet Contrast (Enhanced)
### Issue ID: DAC_Colour_Contrast_Text_Enhanced_01
URL: https://ons-design-system-prototypes.netlify.app/prototypes/dac-may2022-accordions/
Page title: Revised Accordion implementation
Journey: Journey 1, Step 1

Links on the service have a contrast that is greater than the minimum contrast ratio of AA, however, there are some users that may remain to have difficulties seeing the text. Users have a better experience if text contrast meets the AAA criteria.
Foreground: #206095
Background: #FFFFFF
The contrast ratio is: 6.6:1
1.4.6 Contrast (Enhanced) (AAA)
Pass for large text only. Fail for regular text

### Low vision comments: 
“When viewing this page, I struggled to read the light blue text on the link below the heading ‘Revised accordion implementation (April 2022)’ this affects me because of my eye condition I the light blue text fades into the white background therefore prevents me from reading the text on the link . Although it passes the WCAG guidelines for minimum contrast (AA) for regular text. it would benefit me that the text is displayed in a darker colour with a bold font to make it standout more clearly to read.”

### Solution:
We advise reaching for the AAA criteria of colour contrast text enhanced. The minimum contrast ratios for the AAA criteria greatly enhance the readability of the page for low vision users.

**Comment by jrbarnes9** [comment]
Blocked - waiting on response from brand design team

**Comment by jrbarnes9** [comment]
> DS’ colour combinations pass the link contrast checker. https://webaim.org/resources/linkcontrastchecker/
It looks like the logic was to increase the brightness was to aid contrast with more variables at play than just the background colour.





> [11:01](https://onsdigital.slack.com/archives/C03T2241E06/p1660212062469949)
For usability and accessibility, links should be underlined by default. Otherwise, [link text must have at least 3:1 contrast with surrounding body text](https://webaim.org/articles/contrast/#only), and must present a non-color indicator (typically underline) on mouse hover and keyboard focus.
In addition, both links and body text must have at least 4.5:1 contrast with the background (3:1 for large text) to meet WCAG 2 Level AA.


> Rachel Price
  [11:05 AM](https://onsdigital.slack.com/archives/C03T2241E06/p1660212321894049)
Yep their black and blue combo are above 3:1 contrast against each other, and the blue is AA compliant against white (5.19:1)
[11:06](https://onsdigital.slack.com/archives/C03T2241E06/p1660212367342189)
Whereas ours is 2.4:1 blue against black, and the blue is 6.62:1 against white
[11:06](https://onsdigital.slack.com/archives/C03T2241E06/p1660212373153839)
So they've gone the other way with it


> Jon Barnes
  [11:10 AM](https://onsdigital.slack.com/archives/C03T2241E06/p1660212618565499)
so are they saying it’s more important to have the contrast against the black because paragraph links are underlined anyway?
[11:11](https://onsdigital.slack.com/archives/C03T2241E06/p1660212676166139)
So trying to make just the colour AAA against white would have a detrimental effect?


> Grace Ellins
  [11:19 AM](https://onsdigital.slack.com/archives/C03T2241E06/p1660213163626839)
I think it’s as important to have contrast between the black and blue, even though they don’t overlap as you need to distinguish a difference and show there’s a potential action for the user to take. (edited) 
[11:19](https://onsdigital.slack.com/archives/C03T2241E06/p1660213190642869)
Technically underlining a hyperlink is the difference, but the guidance says you can do even more to make it better


> Jon Barnes
  [11:28 AM](https://onsdigital.slack.com/archives/C03T2241E06/p1660213692324019)
I guess then though you’re only dealing with AA compliance, so those who need contrast at AAA level are solely relying on the underline
[11:28](https://onsdigital.slack.com/archives/C03T2241E06/p1660213707937949)
so it’s a trade-off


> Rachel Price
  [11:35 AM](https://onsdigital.slack.com/archives/C03T2241E06/p1660214121693789)
Wonder if it's worth chatting to GDS about why they made that decision and whether they had any issues with it? To see what testing they did and if DAC have flagged anything to them before

**Comment by emilymaylawes** [comment]
I don't think we can achieve this to meet both AAA 7:1 contrast requirements and AA non-text 3:1 contrast/use of colour without an outline. Playing around with the minutae of the hex codes, I can only achieve 2.99 with black and 7:1 with white, or 3:1 with black and 6.99:1 with white.

This is also from darkening the black to #000 from #222 which is the brand colour, so deviating off brand and text readability.

We already provide the underline, passing use of colour, so perhaps it is easiest to darken the blue ever so slightly to achieve AAA?

**Comment by JNATION85** [comment]
Need to speak with Brand team to understand what's needed

**Comment by Majsol** [comment]
If we make the current link colour (Ocean blue #206095) 4% darker it will pass AAA accessibility. So change the colour to #1F5C8F with the contrast ratio 7.03:1 (the minimum is 7:1)

https://webaim.org/resources/contrastchecker/?fcolor=206095&bcolor=FFFFFF

**Comment by daniellecorke** [comment]
As this is Level AAA and there is no good solution, we will close this ticket.
